macosx/Controller.m

```diff
@@ -853,6 +859,41 @@ - (void) applicationWillTerminate: (NSNotification *) notification
     tr_sessionClose(fLib);
 }
 
+- (void) setupTouchBarIfReady
+{
+    if (touchbar != nil)
+    {
+        [touchbar setDefaultItemIdentifiers: @[createTouchBarItem.identifier,
+                                               openFileTouchBarItem.identifier,
+                                               removeTouchBarItem.identifier,
+                                               NSTouchBarItemIdentifierFixedSpaceSmall,
+                                               pauseAllTouchBarItem.identifier,
+                                               resumeAllTouchBarItem.identifier,
+                                               NSTouchBarItemIdentifierFlexibleSpace,
+                                               shareTouchBarButton.identifier
+                                               ]];
+        [touchbar setCustomizationAllowedItemIdentifiers: @[createTouchBarItem.identifier,
+                                                            openFileTouchBarItem.identifier,
+                                                            pauseAllTouchBarItem.identifier,
+                                                            removeTouchBarItem.identifier,
+                                                            resumeAllTouchBarItem.identifier,
+                                                            speedLimitTouchBarItem.identifier,
+                                                            cleanupTouchBarItem.identifier,
+                                                            quickLookTouchBarItem.identifier,
+                                                            showInfoTouchBarItem.identifier,
+                                                            NSTouchBarItemIdentifierFlexibleSpace,
+                                                            NSTouchBarItemIdentifierFixedSpaceLarge,
+                                                            NSTouchBarItemIdentifierFixedSpaceSmall,
+                                                            shareTouchBarButton.identifier
+                                                            ]];
+    }
+    else
+    {
+        [NSTimer scheduledTimerWithTimeInterval:0.1 target:self selector: @selector(setupTouchBarIfReady) userInfo:nil repeats:NO];
```

Comment:
I'm pretty sure that method exists whether or not the Touch Bar is present. There might be some confusion because there were two builds of macOS at the time, one with the API and one without, but they were unified in the following point release. Also this method isn't intended to check for the presence of a hardware touch bar, rather it's supposed to help avoid doing things to a nil object. I think I remember having atypical trouble nailing down exactly when that IBOutlet is set. **That being said, I agree there's gotta be a better way to do this.** But it's worth keeping in mind that the way the Touch Bar API is structured, it seems clear Apple is planning on Bluetooth keyboards with touchbars at some point. So a one-time check at app launch would defeat that purpose. I have to think about this one…